### PR TITLE
cmake: detect keyutils if WITH_LIBCEPHFS OR WITH_RBD

### PR DESCRIPTION
**Pull request by tchaikov**

### CODE REVIEW

**Comment by tchaikov** [comment]
@wjwithagen does this change make sense to you? it reverts part of c7c07da5a23be3c8652e9d65b18778199f1383c2

**Comment by wjwithagen** [comment]
@tchaikov @bassamtabbara 
I remember looking at the keyutils internals, and porting in its full scale to FreeBSD is going to take quite some lifting. So for the time being it is not going to happen. There is a version in the Linux compat modules, but in how far that is actually usable in a FreeBSD code.
And I do want to start looking at RBD stuff, but then requiring keyutils is going to be a showstopper.

**Comment by tchaikov** [comment]
@wjwithagen okay, since keyutils is still required by rbd, let's worry about it once you decide to look at it.

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -876,11 +868,10 @@ endif(WITH_LIBCEPHFS)
 
 add_subdirectory(journal)
 
-if(LINUX)
-  add_library(krbd_objs OBJECT krbd.cc)
-endif()
-
 if(${WITH_RBD})
```
Comment (bassamtabbara) [inline src/CMakeLists.txt]: should this be `if (WITH_RBD)`

Comment (wjwithagen) [inline src/CMakeLists.txt]: @bassamtabbara 
Yup, that is the normal invocation in the project, yours would work also in this specific case but still without `${}` is preferred. 
But if you want to know, loks up variable in the cmake manual

Comment (tchaikov) [inline src/CMakeLists.txt]: thanks, will change it next time when i happen to be changing this part.

**Pull request merged**